love/src/components/ESS/AuxTel/AuxTelESS.container.jsx

```diff
@@ -99,30 +100,67 @@ const getGroupNames = (salindexList, option) => {
   return [...new Set(subscriptions)];
 };
 
+const prevParseToArraySensors = (prevParse, option) => {
+  const list = Object.values(prevParse[option] ?? {}) ?? [];
+  const objs = Object.values(list).map((values) => {
+    return Object.values(values);
+  });
+
+  const result = Array.prototype.concat.apply([], objs);
+  const sorted = result.sort((a, b) => {
+    return a.sensorName > b.sensorName ? 1 : (
+      a.sensorName < b.sensorName ? -1 : (
+        a.sensorName === b.sensorName && a.indexArr > b.indexArr ? 1 : (
+          a.indexArr < b.indexArr ? -1 : 0
+        )
+      )
+    );
+  });
+  return sorted;
+}
+
+let prevParse = {};
 const parse = (streams, option) => {
+  let arr;
   switch (option) {
     case 'temperature':
-      return parseTemperature(streams);
+      arr = parseTemperature(streams);
+      break;
     case 'relativeHumidity':
-      return parseHumidity(streams);
+      arr = parseHumidity(streams);
+      break;
     case 'airFlow':
-      return parseAirflow(streams);
+      arr = parseAirflow(streams);
+      break;
     case 'airTurbulence':
-      return parseAirTurbulence(streams);
+      arr = parseAirTurbulence(streams);
+      break;
     default:
-      return [];
+      arr = [];
   }
+  arr.forEach((parse) => {
+    if (!prevParse[option]) prevParse[option] = {};
+    if (!prevParse[option][parse.sensorName]) prevParse[option][parse.sensorName] = {};
+    if (!prevParse[option][parse.sensorName][parse.indexArr]) prevParse[option][parse.sensorName][parse.indexArr] = {};
+    prevParse[option][parse.sensorName][parse.indexArr] = parse;
+  })
+  const arrSensors = prevParseToArraySensors(prevParse, option);
+  return arrSensors;
 };
 
 const parseTemperature = (streams) => {
   const temperatures = [];
   streams.forEach((stream) => {
     Object.entries(stream).forEach((entry) => {
       const essData = entry[1];
-      const sensorName = essData?.sensorName.value ?? '';
-      const numChannels = essData?.numChannels.value ?? 1;
-      const values = essData?.temperature.value?.slice(0, numChannels) ?? [];
-      const locations = essData?.location.value.split(',')?.slice(0, numChannels) ?? [];
+      const sensorName = essData?.sensorName?.value ?? '';
+      const numChannels = essData?.numChannels?.value ?? 1;
+      const values = essData?.temperature?.value?.slice(0, numChannels) ?? [];
+      const locations = essData?.location?.value.split(',')?.slice(0, numChannels) ?? [];
+      const timestamp = essData?.timestamp?.value ?? undefined;
+      const xPosition = essData?.xPosition?.value ?? undefined;
+      const yPosition = essData?.yPosition?.value ?? undefined;
+      const zPosition = essData?.zPosition?.value ?? undefined;
```

Comment:
I think the `?? undefined` at the end is not necessary as it will be reached if `essData?.timestamp?.value` is `undefined` right?
```suggestion
      const timestamp = essData?.timestamp?.value;
      const xPosition = essData?.xPosition?.value;
      const yPosition = essData?.yPosition?.value;
      const zPosition = essData?.zPosition?.value;
```